libs/database-entity/src/dto.rs

```diff
@@ -1378,9 +1378,45 @@ pub struct TemplateMinimal {
   pub is_featured: bool,
 }
 
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct TemplateMinimalWithPublishInfo {
+  pub view_id: Uuid,
+  pub created_at: DateTime<Utc>,
+  pub last_updated_at: DateTime<Utc>,
+  pub name: String,
+  pub description: String,
+  pub view_url: String,
+  pub creator: TemplateCreatorMinimal,
+  pub categories: Vec<TemplateCategoryMinimal>,
+  pub is_new_template: bool,
+  pub is_featured: bool,
+  pub publish_info: PublishInfo,
+}
+
+impl TemplateMinimalWithPublishInfo {
+  pub fn from_template_and_publish_info(
+    template: &TemplateMinimal,
```

Comment:
Can the params be owned instead? Using a ref to `Template`Minimal` and `PublishInfo` seem to have a lot of cloning. I'm also wondering why not just impl the `std::convert::From` for `(TemplateMinimal, PublishInfo)` tuple type.